test/Conversion/amd/amdgpu_membar.mlir

```diff
@@ -93,4 +93,156 @@ tt.func @pipelined_async_copy_local_to_global_3(%A: !tt.ptr<f16>, %B: !tt.ptr<f1
   // CHECK: tt.return
   tt.return
 }
+
+// Check the we do not get a barrier for LocalLoad if the token comes from a previous loop iteration
+tt.func @async_wait_in_previous_loop_iteration(%a_ptr: tensor<16x16x!tt.ptr<f16>, #AL>, %loopIterCount: i32) {
```

Comment:
Add `// CHECK-LABEL` to separate sections. Similarly for all new tests in the following.